react-carousel-component/src/carousel.jsx

```diff
@@ -9,24 +9,24 @@ class Carousel extends React.Component {
     this.arrowClicks = this.arrowClicks.bind(this);
     this.renderProgressDots = this.renderProgressDots.bind(this);
     this.clickDots = this.clickDots.bind(this);
+    this.timerId = setInterval(this.updateCarousel, 3000);
```

Comment:
It's best to start timers in `componentDidMount` (directly or indirectly by calling another method).